TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/velocity/managers/LobbyManager.java

```diff
@@ -0,0 +1,125 @@
+package cloud.timo.TimoCloud.velocity.managers;
+
+import cloud.timo.TimoCloud.api.TimoCloudAPI;
+import cloud.timo.TimoCloud.api.objects.ServerGroupObject;
+import cloud.timo.TimoCloud.api.objects.ServerObject;
+import cloud.timo.TimoCloud.base.TimoCloudBase;
+import cloud.timo.TimoCloud.velocity.TimoCloudVelocity;
+import cloud.timo.TimoCloud.velocity.objects.LobbyChooseStrategy;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ServerConnection;
+import com.velocitypowered.api.proxy.server.RegisteredServer;
+import com.velocitypowered.api.proxy.server.ServerInfo;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LobbyManager {
+
+    private static final long INVALIDATE_CACHE_TIME = 2000;
+
+    private final Map<UUID, List<String>> lobbyHistory;
+    private final Map<UUID, Long> lastUpdate;
+
+    public LobbyManager() {
+        lobbyHistory = new HashMap<>();
+        lastUpdate = new HashMap<>();
+    }
+
+    private List<String> getVisitedLobbies(UUID uuid) {
+        lobbyHistory.putIfAbsent(uuid, new ArrayList<>());
+        lastUpdate.putIfAbsent(uuid, 0L);
+        if (new Date().getTime() - lastUpdate.get(uuid) >= INVALIDATE_CACHE_TIME) {
+            lobbyHistory.put(uuid, new ArrayList<>());
+        }
+        lastUpdate.put(uuid, new Date().getTime());
+        return lobbyHistory.get(uuid);
+    }
+
+    public void addToHistory(UUID uuid, String server) {
+        lobbyHistory.putIfAbsent(uuid, new ArrayList<>());
+        lobbyHistory.get(uuid).add(server);
+    }
+
+    private LobbyChooseStrategy getLobbyChooseStrategy() {
+        return LobbyChooseStrategy.valueOf(TimoCloudVelocity.getInstance().getFileManager().getConfig().getString("LobbyChooseStrategy"));
+    }
+
+    public RegisteredServer searchFreeLobby(UUID uuid, ServerInfo notThis) {
+        ServerGroupObject group = TimoCloudAPI.getUniversalAPI().getServerGroup(TimoCloudVelocity.getInstance().getFileManager().getConfig().getString("fallbackGroup"));
```

Comment:
Please don't just copy such large sections of code but rather create a global util returning the matching lobby.

Reply:
should i change that in the bungeecord too?

Reply:
Yes, please!